### PR TITLE
Support additional setup in ruby pipelines

### PR DESCRIPTION
**Pull request by ryanwi**
use case is a Ruby app that also needs python setup.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -250,7 +255,7 @@ jobs:
   BUILD:
     name: BUILD
     needs: CHECK
-    uses: signalwire/actions-template/.github/workflows/ci-build.yml@main
+    uses: signalwire/actions-template/.github/workflows/ci-build.yml@ryanwi/additional-setup
```
Comment (ryanwi) [inline .github/workflows/ci.yml:258]:
```suggestion
    uses: signalwire/actions-template/.github/workflows/ci-build.yml@main
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -212,7 +216,7 @@ jobs:
 
   CHECK:
     name: CI
-    uses: signalwire/actions-template/.github/workflows/ci-check.yml@main
+    uses: signalwire/actions-template/.github/workflows/ci-check.yml@ryanwi/additional-setup
```
Comment (ryanwi) [inline .github/workflows/ci.yml:219]:
```suggestion
    uses: signalwire/actions-template/.github/workflows/ci-check.yml@main
```

File: .github/workflows/ci-check.yml [hunk 3]
```diff
@@ -152,7 +156,7 @@ jobs:
       uses: actions/checkout@v4
       with:
         repository: signalwire/actions-template
-        ref: main
+        ref: ryanwi/additional-setup
```
Comment (ryanwi) [inline .github/workflows/ci-check.yml:159]:
```suggestion
        ref: main
```

File: .github/workflows/ci-build.yml [hunk 4]
```diff
@@ -224,7 +224,7 @@ jobs:
       uses: actions/checkout@v4
       with:
         repository: signalwire/actions-template
-        ref: main
+        ref: ryanwi/additional-setup
```
Comment (ryanwi) [inline .github/workflows/ci-build.yml:227]:
```suggestion
        ref: main
```

**Comment by Null-Is-Null** [comment]
This will work, but that use of `eval` doesn't feel good from a security perspective. I know impact will be limited since this is running on a runner and will only be used internally, but what if we standardized on something like a `ci-bootstrap.sh` script that lives in the target ruby repo that gets run if present? I'm fine with proceeding with the approach in this PR for now if that's preferable.

**Comment by ryanwi** [comment]
>  use of eval

isn't that what we're already doing here? https://github.com/signalwire/actions-template/blob/main/.github/actions/test-ruby/action.yml#L121

**Comment by Null-Is-Null** [comment]
After reviewing the lines you linked I realize we probably shouldn't be doing that there either. That's not exactly the same as an `eval` but is effectively executing commands passed as a string with some checks in place (seeing if the command starts with bundle). Approving for now since potential impact is limited as described in prior comments.

**Comment by ryanwi** [comment]
That's what I thought you'd say :-)

**Pull request merged**